index.d.ts

```diff
@@ -5,42 +5,75 @@ import { Tags, TagType } from 'prismarine-nbt'
 type ItemLike = Item | null
 
 declare class Item {
-    constructor(type: number, count: number, metadata?: number, nbt?: object);
-    type: number;
-    slot: number;
-    count: number;
-    metadata: number;
-    nbt: Tags[TagType] | null;
-    name: string;
-    displayName: string;
-    stackSize: number;
-    durabilityUsed: number;
-    enchants: NormalizedEnchant[];
-    repairCost: number;
-    customName: string | null;
-    customLore: string | null;
-    readonly spawnEggMobName: string;
-    static equal(item1: Item, item2: Item, matchStackSize: boolean): boolean;
-    static toNotch(item: ItemLike): NotchItem;
-    static fromNotch(item: NotchItem): ItemLike;
-    static anvil (itemOne: ItemLike, itemTwo: ItemLike, creative: boolean, rename: string | undefined): { xpCost: number, item: ItemLike }
+  constructor(type: number, count: number, metadata?: number, nbt?: object, stackId?: number);
+  type: number;
+  slot: number;
+  count: number;
+  metadata: number;
+  nbt: Tags[TagType] | null;
+  stackId: number;
+  name: string;
+  displayName: string;
+  stackSize: number;
+  durabilityUsed: number;
+  get enchants(): { name: string; lvl: number }[];
+  set enchants(enchantments: { name: string; lvl: number }[]);
+  blocksCanPlaceOn: string[];
+  blocksCanDestroy: string[];
+  repairCost: number;
+  customName: string | null;
+  customLore: string | null;
+  readonly spawnEggMobName: string;
+  static equal(item1: Item, item2: Item, matchStackSize: boolean, matchNbt: boolean): boolean;
+  static toNotch(item: ItemLike): NetworkItem;
+  static fromNotch(item: NetworkItem): ItemLike;
+  static anvil(
+    itemOne: ItemLike,
+    itemTwo: ItemLike,
+    creative: boolean,
+    rename: string | undefined
+  ): { xpCost: number; item: ItemLike };
+  static currentStackId: number;
+  static nextStackId(): number;
 }
 
-declare interface NotchItem {
-    // 1.8 - 1.12
-    blockId?: number;
-    itemDamage?: number;
-    // 1.13 - 1.15
-    present?: boolean;
-    itemId?: number;
+declare interface PcNetworkItem {
+  // 1.8 - 1.12
+  blockId?: number;
+  itemDamage?: number;
+  // 1.13 - 1.15
+  present?: boolean;
+  itemId?: number;
 
-    itemCount?: number;
-    nbtData?: Buffer;
+  itemCount?: number;
+  nbtData?: Buffer;
 }
 
-declare interface NormalizedEnchant {
-    name: string;
-    lvl: number
+declare interface BedrockNetworkItem {
```

Comment:
Remove

Reply:
The entire interface?

Reply:
Yes, types should hold stable API things and network packets are not stable in structure (eg new fields may be added)